native/src/components/RemoteContent.tsx

```diff
@@ -73,6 +73,14 @@ const RemoteContent = ({
   const { t } = useTranslation()
   const { width: deviceWidth } = useWindowDimensions()
 
+  const resourceMap: { [url: string]: string } = Object.entries(resourceCache).reduce(
+    (previous, [url, { filePath }]) => ({
+      ...previous,
+      [url]: getStaticServerFileUrl(filePath, staticServerUrl),
+    }),
+    {},
+  )
+
```

Comment:
I think it will create a new object for `...previous` each on each iteration.

Grok said it's complexity O(n²) so it gave me this as alternative (I couldn't comprehend the solution 😅 but looks clean and O(n)):
```js
const resourceMap = Object.fromEntries(
  Object.entries(resourceCache).map(([url, { filePath }]) => [
    url,
    getStaticServerFileUrl(filePath, staticServerUrl)
  ])
)
```

Reply:
Good suggestion, but this actually made me think and realize that we don't even need any object values and just `{ url: path }` is enough, therefore i further reduced the complexity and made this obsolete.